dashboard/app/controllers/maker_controller.rb

```diff
@@ -121,18 +121,26 @@ def login_code
   # GET /maker/display_code
   # renders a page for users to copy and paste a login key
   def display_code
-    # Generate encrypted code to display to user
-    user_auth = current_user.authentication_options.find_by_credential_type(AuthenticationOption::GOOGLE)
-    @secret_code = Encryption.encrypt_string_utf8(
-      Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth['authentication_id'] + user_auth['credential_type']
-    )
+    if current_user
+      # Generate encrypted code to display to user
+      user_auth = current_user.authentication_options.find_by_credential_type(AuthenticationOption::GOOGLE)
+      if user_auth
+        @secret_code = Encryption.encrypt_string_utf8(
+          Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth['authentication_id'] + user_auth['credential_type']
+        )
+      end
+    end
   end
 
   # GET /maker/confirm_login
   # Renders a page to confirm uses want to login via Google OAuth
   # This route is need to convert the GET request from the Maker App into
   # a POST that can be used to login via Google OAuth
   def confirm_login
+    session[:user_return_to] ||= params[:user_return_to]
```

Comment:
i think we always want to set `session[:user_return_to]` here ([`||=`](http://www.rubyinside.com/what-rubys-double-pipe-or-equals-really-does-5488.html) in ruby means it will only be set if the current value is nil or false)